### PR TITLE
backport the buildkite pipelines from the main branch to the branch 7.17

### PR DESCRIPTION
**Pull request by sharbuz**
## What is the problem this PR solves?

backport the buildkite pipelines from the main branch to branch 7.17

## How does this PR solve the problem?

We will able to remove the Jenkins pipelines for the fleet-server repository

## How to test this PR locally

It works in the 8.9 and 8.10 branches

## Related issues

Related to the issue: https://github.com/elastic/fleet-server/issues/2517

### CODE REVIEW

**Review by jlind23 — APPROVED** [review]

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        DOCKER_TAG: "${BUILDKITE_TAG}"
+    if: "build.env('BUILDKITE_TAG') != ''"
+    depends_on:
+      - step: "release-test"
+        allow_failure: false
+
+  - trigger: "fleet-server-package-mbp"
+    label: ":esbuild: Downstream - Package"
+    key: "downstream-package"
+    if: "build.env('BUILDKITE_PULL_REQUEST') == 'false'"
```
Comment (mrodm) [inline .buildkite/pipeline.yml:135]: I think this needs to be udpated as in this PR: https://github.com/elastic/fleet-server/pull/2890

File: catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      name: fleet-server
+      description: Buildkite pipeline for the Elastic Fleet Server project
+    spec:
+      repository: elastic/fleet-server
+      pipeline_file: ".buildkite/pipeline.yml"
+      branch_configuration: "main 8.* 7.* v8.* v7.*"
```
Comment (mrodm) [inline catalog-info.yaml:44]: :+1:

**Review by jlind23 — DISMISSED** [review]

**Review by jlind23 — APPROVED** [review]

**Pull request merged**